json_database/hpm.py

```
@@ -11,9 +11,18 @@ class JsonDB(AbstractDB):
     """HiveMind Database implementation using JSON files."""
     name: str = "clients"
     subfolder: str = "hivemind-core"
+    password: Optional[str] = None
 
     def __post_init__(self):
-        self._db = JsonStorageXDG(self.name, subfolder=self.subfolder, xdg_folder=xdg_data_home())
+        if self.password:
+            self._db = EncryptedJsonStorageXDG(encrypt_key=self.password,
```

Ruff (0.8.2)

Undefined name `EncryptedJsonStorageXDG`

(F821)